### ISSUE TITLE
add seperate select tool?

### ISSUE THREAD

**Issue by dobkeratops** [bbernhard/imagemonkey-core#119]
When annotating with the polygon tool, if you need to draw more than one polygon (disconnected regions, objects with holes etc), clicks are sometimes interpreted as selection of existing polygons (similar to the problem with the rectangle tool);  - perhaps a seperate mode for selection/adjustment would alleviate this. 
Alternatively , If you wanted to retain the existing behaviour, perhaps this could be a toggle (defaulting to on) rather than a seperate tool (ie. the available modes would be rectangle,ellipse,polygon, or  rectangle/select  ellipse/select polygon/select).

It is great that you *can* adjust existing objects, it's just control over when you want to do one or the other would help.  But I do know from my own recent experiment (and seeing other things) the fine line between learning curve, control,versatility.. it could be that vertex/poly tools are just hard to make good for casual users.

The best icon for this would probably be a pointer-arrow with move-arrow combination, similar to photoshop ("move tool"):  

<img width="88" alt="screen shot 2018-05-19 at 08 51 49" src="https://user-images.githubusercontent.com/1120754/40266381-04b11814-5b42-11e8-84c0-8e850462ac97.png">

**Comment by bbernhard** [comment]
Many thanks for pointing that out - that makes definitely sense! 

Would you add that icon to the existing toolbox? I am afraid that at some point the toolbox might become too big? You already mentioned that it might be a good idea to arrange the toolbox from top to bottom and move it to the left side of the screen. I haven't looked into that, but I guess this could indeed be better (especially if we add more and more icons to the toolbox). Another option would be to catch the right mousebutton click event and serve a custom menu with some options (I have seen that a few times in some online image editing applications). But not sure if thats better in terms of user experience?

**Comment by dobkeratops** [comment]
>>Another option would be to catch the right mousebutton click event and serve a custom menu with some options

yes a problem with that is touchscreens, but sometimes they do 'press and hold' for alternate actions; and a right click can always be a shortcut for things the UI shows elsewhere
